infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -128,7 +129,9 @@
     private long maxConflictWildcard = Address.NOT_FOUND;
 
     private final long defaultCacheSize = Long.MAX_VALUE;
-    private final Cache<Integer, Long> conflictToGlobalTailCache;
+    private final long defaultPerStreamCacheSize = Long.MAX_VALUE;
+    private long perStreamCacheSize = defaultPerStreamCacheSize;
+    private final Cache<UUID, Cache<Integer, Long>> conflictToGlobalTailCache;
```

Comment:
Yes, that _could_ be a problem. Fortunately for all Java library objects, .hashCode() is stable across all JVMs...
I think if you wanted to solve this the right way you would add some hook where we could generate the hash code post serialization. That's probably going to require some invasive updates, but I can give it a stab. BTW, Guava's Hasher can only deal with primitive types, as I suspected.